rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -1676,6 +1695,126 @@ rocalVideoFileSource(
 }
 
 RocalTensor ROCAL_API_CALL
+rocalNumpyFileSource(
+    RocalContext p_context,
+    const char* source_path,
+    unsigned shard_count,
+    std::vector<std::string> files,
+    bool is_output,
+    bool shuffle,
+    bool loop,
+    unsigned seed,
+    RocalShardingInfo rocal_sharding_info) {
+    Tensor* output = nullptr;
+    auto context = static_cast<Context*>(p_context);
+    try {
+        auto max_dimensions = evaluate_numpy_data_set(StorageType::NUMPY_DATA, DecoderType::SKIP_DECODE,
+                                                      source_path, files);
+
+        RocalTensorDataType tensor_data_type;
+        std::map<int, RocalTensorDataType> data_type_map = {
+            {0, RocalTensorDataType::FP32},
+            {1, RocalTensorDataType::FP16},
+            {2, RocalTensorDataType::UINT8},
+            {3, RocalTensorDataType::INT8},
+            {4, RocalTensorDataType::UINT32},
+            {5, RocalTensorDataType::INT32},
+        };
+        tensor_data_type = data_type_map[max_dimensions.back()];
+        max_dimensions.pop_back();
+        unsigned num_of_dims = max_dimensions.size() + 1;
+        std::vector<size_t> dims;
+        dims.resize(num_of_dims);
```

Comment:
```suggestion
        unsigned num_of_dims = max_dimensions.size() + 1;
        std::vector<size_t> dims;
        dims.resize(num_of_dims);
```

Reduce these 3 lines as
`std::vector<size_t> dims(max_dimensions.size() + 1);`